src/main/java/seedu/address/logic/parser/EditTaskCommandParser.java

```diff
@@ -0,0 +1,101 @@
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CONTACT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ENDTIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STARTTIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.EditTaskDescriptor;
+import seedu.address.logic.commands.EditTaskCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.name.Name;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new EditCommand object
+ */
+public class EditTaskCommandParser implements Parser<EditTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditTaskCommand
+     * and returns an EditTaskCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditTaskCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DATE, PREFIX_STARTTIME,
+                        PREFIX_ENDTIME, PREFIX_TAG, PREFIX_CONTACT);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditTaskCommand.MESSAGE_USAGE), pe);
+        }
+
+        EditTaskDescriptor editTaskDescriptor = new EditTaskDescriptor();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            editTaskDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_DATE).isPresent()) {
+            editTaskDescriptor.setDate(ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get()));
+        }
+        if (argMultimap.getValue(PREFIX_STARTTIME).isPresent()) {
+            editTaskDescriptor.setStartTime(ParserUtil.parseStartTime(argMultimap.getValue(PREFIX_STARTTIME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_ENDTIME).isPresent()) {
+            editTaskDescriptor.setEndTime(ParserUtil.parseEndTime(argMultimap.getValue(PREFIX_ENDTIME).get()));
+        }
+        parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editTaskDescriptor::setTags);
+        parsePersonsForEdit(argMultimap.getAllValues(PREFIX_CONTACT)).ifPresent(editTaskDescriptor::setPersons);
```

Comment:
I think you might be missing steps to check if the edited persons are valid i.e. is it in the contact list?

Reply:
Thanks! ended up added those changed to the EditTaskCommand class instead of the parser